File: greenDaoSample/src/main/java/com/hzy/greendao/MainActivity.java
package com.hzy.greendao;


import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;


import com.ashokvarma.bottomnavigation.BottomNavigationBar;
import com.ashokvarma.bottomnavigation.BottomNavigationItem;
import com.hzy.greendao.adapter.PagerAdapter;
import com.hzy.greendao.fragment.BlankFragment1;
import com.hzy.greendao.fragment.BlankFragment2;


import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import java.util.ArrayList;
import java.util.List;

import pl.com.salsoft.sqlitestudioremote.SQLiteStudioService;

public class MainActivity extends AppCompatActivity implements BottomNavigationBar
        .OnTabSelectedListener, ViewPager.OnPageChangeListener{
    BottomNavigationBar bottomNavigationBar;
    List<Fragment> fragments;
    ViewPager viewPager;
    int currentPos = 0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViewPager();
        initBottomNavi();
        initData();
       // save();
       // query();
    }

    private void initViewPager() {
        viewPager =  findViewById(R.id.viewPager);
        fragments = new ArrayList<>();
        fragments.add(new BlankFragment1());
        fragments.add(new BlankFragment2());
        viewPager.setAdapter(new PagerAdapter(getSupportFragmentManager(), fragments));
        viewPager.addOnPageChangeListener(this);
        viewPager.setCurrentItem(0);
    }

    private void initBottomNavi() {
        bottomNavigationBar = findViewById(R.id.bottom_navigation_bar);
        bottomNavigationBar.setTabSelectedListener(this);
        bottomNavigationBar.setMode(BottomNavigationBar.MODE_FIXED);
        bottomNavigationBar.setBackgroundStyle(BottomNavigationBar.BACKGROUND_STYLE_STATIC);
        bottomNavigationBar.setBarBackgroundColor(R.color.white);
        bottomNavigationBar
                .addItem(new BottomNavigationItem(R.drawable.home_icon, "待办").setActiveColorResource(R.color.blue))
                .addItem(new BottomNavigationItem(R.drawable.sys_set_icon, "工作台").setActiveColorResource(R.color.blue))
                .setFirstSelectedPosition(0)
                .initialise();
    }

    private void initData() {

    }


    @Override
    public void onPageScrolled(int i, float v, int i1) {

    }

    @Override
    public void onPageSelected(int i) {
//        Log.e("MainActivity","onPageSelected"+i);
        if(currentPos != i){
            bottomNavigationBar.selectTab(i);
        }
    }

    @Override
    public void onPageScrollStateChanged(int i) {

    }

    @Override
    public void onTabSelected(int position) {
//        Log.e("MainActivity","onTabSelected"+position);
        currentPos = position;
         viewPager.setCurrentItem(position);

    }

    @Override
    public void onTabUnselected(int position) {
//        Log.e("MainActivity","onTabUnselected"+position);
    }

    @Override
    public void onTabReselected(int position) {
//        Log.e("MainActivity","onTabReselected"+position);
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void onMessageEvent(String  o) {

    }
    @Override
    public void onStart() {
        if (!EventBus.getDefault().isRegistered(this)) {
            EventBus.getDefault().register(this);
        }
        super.onStart();

    }

    @Override
    public void onStop() {
        super.onStop();
        if (EventBus.getDefault().isRegistered(this)) {
            EventBus.getDefault().unregister(this);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        SQLiteStudioService.instance().stop();

    }
}

File: baseFrameKotlin/src/main/java/com/hzy/smartfield/MainActivity.kt
package com.hzy.smartfield


import android.view.View
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentTransaction
import com.ashokvarma.bottomnavigation.BottomNavigationBar
import com.ashokvarma.bottomnavigation.BottomNavigationItem
import com.hzy.smartfield.base.BaseActivity
import com.hzy.smartfield.fragment.CustomerFragment
import com.hzy.smartfield.fragment.DaiBanFragment
import com.hzy.smartfield.fragment.HousingFragment
import com.hzy.smartfield.fragment.MineFragment
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.base_title_bar.*

class MainActivity : BaseActivity(), BottomNavigationBar.OnTabSelectedListener {


    private var fragmentManager: FragmentManager? = null
    private var transaction: FragmentTransaction? = null
    private var daiBanFragment: DaiBanFragment? = null
    private var customerFragment: CustomerFragment? = null
    private var housingFragment: HousingFragment? = null
    private var mineFragment: MineFragment? = null
    override fun setLayoutResourceID(): Int = R.layout.activity_main

    override fun init() {}

    override fun initView() {
        ll_title_bar.visibility = View.GONE;
        fragmentManager = supportFragmentManager
        bottomNavigationBar.setMode(BottomNavigationBar.MODE_FIXED)
        bottomNavigationBar
            .addItem(BottomNavigationItem(R.mipmap.icon_daiban, "待办"))
            .addItem(BottomNavigationItem(R.mipmap.icon_customer, "客户"))
            .addItem(BottomNavigationItem(R.mipmap.icon_housing, "房源"))
            .addItem(BottomNavigationItem(R.mipmap.icon_mine, "我的"))
            .setActiveColor("#2D82FF")
            .setFirstSelectedPosition(0)
            .setBarBackgroundColor(R.color.white)
            .initialise()
        bottomNavigationBar.setTabSelectedListener(this)
        initDaiBanFragment()
    }


    override fun initData() {
    }

    private fun initDaiBanFragment() {
        transaction = fragmentManager!!.beginTransaction()
        if (daiBanFragment == null) {
            daiBanFragment = DaiBanFragment()
            transaction!!.add(R.id.container, daiBanFragment!!)
        }
        //隐藏所有fragment
        hideFragment(transaction!!)
        //显示待办fragment
        transaction!!.show(daiBanFragment!!)
        transaction!!.commit()
    }

    private fun initCustomerFragment() {
        transaction = fragmentManager!!.beginTransaction()
        if (customerFragment == null) {
            customerFragment = CustomerFragment()
            transaction!!.add(R.id.container, customerFragment!!)
        }
        //隐藏所有fragment
        hideFragment(transaction!!)
        //显示客户fragment
        transaction!!.show(customerFragment!!)
        transaction!!.commit()
    }

    private fun initHousingFragment() {
        transaction = fragmentManager!!.beginTransaction()
        if (housingFragment == null) {
            housingFragment = HousingFragment()
            transaction!!.add(R.id.container, housingFragment!!)
        }
        //隐藏所有fragment
        hideFragment(transaction!!)
        //显示房源fragment
        transaction!!.show(housingFragment!!)
        transaction!!.commit()
    }

    private fun initMineFragment() {
        transaction = fragmentManager!!.beginTransaction()
        if (mineFragment == null) {
            mineFragment = MineFragment()
            transaction!!.add(R.id.container, mineFragment!!)
        }
        //隐藏所有fragment
        hideFragment(transaction!!)
        //显示我的fragment
        transaction!!.show(mineFragment!!)
        transaction!!.commit()
    }

    //隐藏所有的fragment
    private fun hideFragment(transaction: FragmentTransaction) {
        if (daiBanFragment != null) {
            transaction.hide(daiBanFragment!!)
        }
        if (customerFragment != null) {
            transaction.hide(customerFragment!!)
        }
        if (housingFragment != null) {
            transaction.hide(housingFragment!!)
        }
        if (mineFragment != null) {
            transaction.hide(mineFragment!!)
        }
    }

    override fun onTabReselected(position: Int) {
    }

    override fun onTabUnselected(position: Int) {
    }


    override fun onTabSelected(position: Int) {
        when (position) {
            0 -> initDaiBanFragment()
            1 -> initCustomerFragment()
            2 -> initHousingFragment()
            3 -> initMineFragment()
            else -> {

            }
        }
    }
}
